packages/x-adapter-platform/src/schemas/facet.schema.ts

```diff
@@ -0,0 +1,84 @@
+import { createMutableSchema, Schema } from '@empathyco/x-adapter-next';
+import {
+  EditableNumberRangeFacet,
+  FacetModelName,
+  HierarchicalFacet,
+  HierarchicalFilter,
+  NumberRangeFacet,
+  SimpleFacet
+} from '@empathyco/x-types';
+import { PlatformFacet, PlatformHierarchicalFilter } from '../types';
+import { numberFilterMutableSchema } from './filters/number-filter.schema';
+import { simpleMutableFilterSchema } from './filters/simple-filter.schema';
+
+export const facetSchema: Schema<
+  PlatformFacet,
+  HierarchicalFacet | NumberRangeFacet | SimpleFacet | EditableNumberRangeFacet
+> = {
+  id: 'facet',
+  label: 'facet',
+  modelName: ({ facet }) => getFacetType(facet) as any,
+  filters: {
+    $path: 'values',
+    $subSchema: ({ facet }) => getFilterSchemaFromFacetType(facet),
+    $context: {
+      parentId: 'facet'
+    }
+  }
+};
+
+export const facetMutableSchema = createMutableSchema(facetSchema);
+
+export const hierarchicalFilterSchema: Schema<PlatformHierarchicalFilter, HierarchicalFilter> = {
+  facetId: 'filter',
+  label: 'value',
+  id: 'filter',
+  totalResults: 'count',
+  parentId: (_, $context) => $context?.parentId as string,
+  selected: () => false,
+  modelName: () => 'HierarchicalFilter',
+  children: {
+    $path: 'children',
+    $subSchema: facetMutableSchema as any
+  }
+};
+
+export const hierarchicalFilterMutableSchema = createMutableSchema(hierarchicalFilterSchema);
```

Comment:
Can you move this `schema` to its own file? Same approach as with the other type of filters.

Reply:
Since `facetSchema` is depending on `hierarchicalFilterSchema` and viceversa I think makes sense to keep them together. Besides, I did not find an easy way to split them and make it work, to be honest 😅